File: 3634-minimum-removals-to-balance-array/3634-minimum-removals-to-balance-array.py
class Solution(object):
    def minRemoval(self, nums, k):
        nums.sort()
        n = len(nums)
        maxlen = 1
        j =0
        for i in range(n):
            while j < n and nums[j] <=nums[i]*k:
                j += 1
            cur = j -i
            if cur > maxlen:
                maxlen = cur
        
        return n - maxlen
        """
        :type nums: List[int]
        :type k: int
        :rtype: int
        """
        

File: 3634-minimum-removals-to-balance-array/3634-minimum-removals-to-balance-array.cpp
class Solution {
public:
    int minRemoval(vector<int>& nums, int k) {
        int n = nums.size();
        
        sort(nums.begin(),nums.end());
        int maxLen = 1;
        int j =0;
        for(int i = 0; i < n; i++){
            while(j < n && (long long)nums[j] <= (long long)nums[i]*k){
                j++;
            }
            int current = j - i;
            maxLen = max(maxLen,current);
        }
        return n - maxLen;
    }
};